lib/ansible/modules/system/filesystem.py

```diff
@@ -213,56 +224,46 @@ def main():
     rc,raw_fs,err = module.run_command("%s -c /dev/null -o value -s TYPE %s" % (cmd, dev))
     fs = raw_fs.strip()
 
-    if fs == fstype and resizefs is False and not force:
-        module.exit_json(changed=False)
-    elif fs == fstype and resizefs is True:
-        # Get dev and fs size and compare
-        devsize_in_bytes = _get_dev_size(dev, module)
-        fssize_in_bytes = _get_fs_size(fssize_cmd, dev, module)
-        if fssize_in_bytes < devsize_in_bytes:
-            fs_smaller = True
-        else:
-            fs_smaller = False
-
-
-        if module.check_mode and fs_smaller:
-            module.exit_json(changed=True, msg="Resizing filesystem %s on device %s" % (fstype,dev))
-        elif module.check_mode and not fs_smaller:
-            module.exit_json(changed=False, msg="%s filesystem is using the whole device %s" % (fstype, dev))
-        elif fs_smaller:
-            cmd = module.get_bin_path(growcmd, required=True)
-            rc,out,err = module.run_command("%s %s" % (cmd, dev))
-            # Sadly there is no easy way to determine if this has changed. For now, just say "true" and move on.
-            #  in the future, you would have to parse the output to determine this.
-            #  thankfully, these are safe operations if no change is made.
-            if rc == 0:
-                module.exit_json(changed=True, msg=out)
-            else:
-                module.fail_json(msg="Resizing filesystem %s on device '%s' failed"%(fstype,dev), rc=rc, err=err)
-        else:
-            module.exit_json(changed=False, msg="%s filesystem is using the whole device %s" % (fstype, dev))
-    elif fs and not force:
-        module.fail_json(msg="'%s' is already used as %s, use force=yes to overwrite"%(dev,fs), rc=rc, err=err)
-
-    ### create fs
-
-    if module.check_mode:
-        changed = True
-    else:
+    if not fs or force:
+        if module.check_mode:
+            module.exit_json(changed=True)
         mkfs = module.get_bin_path(mkfscmd, required=True)
-        cmd = None
-
+        if force_flag is None:
+            force_flag = ''
         if opts is None:
-            cmd = "%s %s '%s'" % (mkfs, force_flag, dev)
+            opts = ''
+        if fstype == 'vfat':
+            rc,out,err = module.run_command("%s -s %d %s" % (cmd, devsize_in_bytes, dev))
         else:
-            cmd = "%s %s %s '%s'" % (mkfs, force_flag, opts, dev)
-        rc,_,err = module.run_command(cmd)
+            rc,out,err = module.run_command("%s %s" % (cmd, dev))
```

Comment:
This should probably be `mkfs` also instead of `cmd`.  Additionally, we've lost the `force_flag` and `opts` for some reason.   

How does this look to you?
```diff
@@ -232,10 +232,7 @@ def main():
             force_flag = ''
         if opts is None:
             opts = ''
-        if fstype == 'vfat':
-            rc,out,err = module.run_command("%s -s %d %s" % (cmd, devsize_in_bytes, dev))
-        else:
-            rc,out,err = module.run_command("%s %s" % (cmd, dev))
+        rc,out,err = module.run_command("%s %s %s '%s'" % (mkfs, force_flag, opts, dev))
         if rc == 0:
             module.exit_json(changed=True)
         else:
```